### DISCUSSION TITLE
How can we use ProtonVPN and Rethink App at the same time?

### DISCUSSION THREAD

**Discussion by matt873** [celzero/rethink-app#940] [Q&A] [answered]
Is there any workaround to do this?
Also, is this something to be added as a feature to use both VPN and Rethink App at the same time?

**Comment by iulko** [comment]
With wireguard now available in the app, its possible

**Reply by attab614-cyber** [reply]
يمكن ذلك 
استخدم راوتر يدعم تشغيل الvpn 
ثم استخدم التطبيق الذي تريده على هاتفك

**Reply by attab614-cyber** [reply]
How can we use ProtonVPN and Rethink App at the same time?
Yes 
You can do this using a router like the ASUS RT-AX5400, which allows you to install a built-in VPN service in the settings. This allows you to install any application you want that requires a local or external virtual network. I actually use this method, and it works.

**Comment by ignoramous** [comment, marked as answer]
1. On your Android, visit `https://account.protonvpn.com/downloads`
2. Scroll to WireGuard
   - Give it a name
   - Select Android
   - Enable VPN options as approp
   - Press *Create* or tap on *Download*.
3. Open Rethink, goto *Configure* -> *Proxy* -> *Setup WireGuard*
   - Tap on the "+" *button*
   - Choose *Import*
   - Select the downloaded configuration file
4. If the import is successful, you should see a new profile added to the list.
5. (optional) If you're using the *Advanced* mode (and not *Simple*), tap on that newly added entry.
   - In the new screen that opens up, tap on `Add / Remove applications`. Note: *Simple* mode, by default, routes all apps (except for the ones marked `Bypass from all proxies` from *Configure* -> *Apps* UI).
   - Select apps that should be part of this particular WireGuard profile.
   - Go back.
6. Enable / disable that profile as desirable.

From this discussion on our subreddit: https://www.reddit.com/r/rethinkdns/comments/163e83l/proton_vpn/

**Reply by error-reporting** [reply]
is the dns field in rethink wireguard config supports ipv6 addresses

**Reply by ignoramous** [reply]
IPv6 DNS? Don't see why not.
